src/Tokenizer/Tokens.php

```diff
@@ -295,10 +295,18 @@ public function setSize($size): bool
      */
     public function offsetUnset($index): void
     {
-        $this->changed = true;
-        $this->namespaceDeclarations = null;
         if (isset($this[$index])) {
+            if (isset($this->blockStartCache[$index])) {
```

Comment:
I decided to put it in, but it's dead code - we do not use it in our codebase (except single test case where we test this), but maybe some custom fixers are. let's have it behaving properly